crates/iota-data-ingestion-core/src/worker_pool.rs

```diff
@@ -48,22 +61,27 @@ impl<W: Worker + 'static> WorkerPool<W> {
         let mut idle: BTreeSet<_> = (0..self.concurrency).collect();
         let mut checkpoints = VecDeque::new();
 
-        let mut join_handles = vec![];
+        let mut workers_join_handles = vec![];
 
         // spawn child workers
         for worker_id in 0..self.concurrency {
```

Comment:
Nit: I would suggest running this loop into a dedicated function if possible to make it easier to read and understand the implementation.

Reply:
[fixed](https://github.com/iotaledger/iota/pull/4648/commits/f2d9f8da3328c5aae5815778d69daba4cdbd5bc9#diff-dfef1eeaea282ac0a41b60863611b2244808890824a073bf2264ca6fc775a624R162-R220)